### PR TITLE
chore: add base theme

### PR DESCRIPTION
**Pull request by loido**

### CODE REVIEW

File: src/theme/base/opacity.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const opacity = {
+  0: 0,
```
Comment (TranQuangPhi) [inline src/theme/base/opacity.ts:2]: I think the key should be a string rather than a number

File: src/theme/base/shadows.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
+const shadows = {
+  0: {
```
Comment (TranQuangPhi) [inline src/theme/base/shadows.ts:2]: same the above

File: src/theme/base/borderWidths.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  tiny: 2,
+  small: 4,
+  medium: 6,
+  large: 8,
+  huge: 10,
+  enormous: 12,
+  massive: 14,
+  gargantuan: 16,
+  colossal: 18,
+  largest: 20,
```
[the remark this answers is not in the document]
Comment (loido) [reply src/theme/base/borderWidths.ts]: done

File: src/theme/base/opacity.ts (hunk 1 above)
Comment (loido) [reply src/theme/base/opacity.ts]: done

File: src/theme/base/shadows.ts (hunk 2 above)
Comment (loido) [reply src/theme/base/shadows.ts]: done

File: src/theme/base/opacity.ts (hunk 1 above)
Comment (TranQuangPhi) [reply src/theme/base/opacity.ts]: I don't think terms like tiny, miniature, or petite are appropriate in relation to opacity.

What do you think about:

Light
Bright
Radiant
Shimmering
Gleaming
Dazzling
Shining
Glimmering
Illuminated
Twilight
Gloomy
Somber
Dark

Comment (loido) [reply src/theme/base/opacity.ts]: done

File: src/theme/base/shadows.ts (hunk 2 above)
Comment (TranQuangPhi) [reply src/theme/base/shadows.ts]: same the above

**Review by baonguyenhsts — APPROVED** [review]

**Review by TranQuangPhi — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]
Look good

File: src/core/BaseProvider.tsx [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-empty-interface */
```
Comment (ThinhKimVo) [inline src/core/BaseProvider.tsx:2]: should we remove these lint ignore?

**Review by ThinhKimVo — APPROVED** [review]

**Pull request merged**